app/components/Bargraph/index.tsx

```diff
@@ -0,0 +1,128 @@
+import {
+  ResponsiveContainer,
+  Tooltip,
+  XAxis,
+  YAxis,
+  BarChart,
+  Bar,
+} from "recharts";
+import { addCommas, CustomTooltip, CustomizedLabel } from "../Utility";
+
+export const Bargraph = ({
+  coinHistoryHour,
+  coinCompareHour,
+  load,
+  coinCompare,
+  coinHistory,
+  compare,
+  today,
+}: any) => {
+  let totalVolume = 0;
+  const pdata1 = coinHistoryHour.map((hour: any, i: number) => {
+    const value = hour.QUOTE_VOLUME;
+    totalVolume = totalVolume + value;
+    const valueProper = addCommas(value);
+    return {
+      coin: hour.INSTRUMENT,
+      name: new Date(Date.now() + i * 3600000 - 86400000 + 3600000)
+        .toString()
+        .split("G")[0],
+      value: value,
+      valueProper: valueProper,
+    };
+  });
+
+  let totalVolumeCompare = 0;
+  const pdata1Compare = coinCompareHour.map((hour: any, i: number) => {
+    const value = hour.QUOTE_VOLUME;
+    totalVolumeCompare = totalVolumeCompare + value;
+    const valueProper = addCommas(value);
+    return {
+      coin: hour.INSTRUMENT,
+      name: new Date(Date.now() + i * 3600000 - 86400000 + 3600000)
+        .toString()
+        .split("G")[0],
+      value: value,
+      valueProper: valueProper,
+      index: i + 1,
+    };
+  });
+
+  const combi1 = pdata1.map((data: any, index: number) => {
+    if (pdata1Compare.length) {
+      return {
+        ...data,
+        valueComp: pdata1Compare[index].value,
+        valueCompProper: pdata1Compare[index].valueProper,
+      };
+    } else {
+      return data;
+    }
+  });
```

Comment:
or instead of having 3 maps, maybe somehow combine this in an algo outside this component?